common/src/main/java/com/wynntils/models/characterstate/CharacterStateModel.java

```diff
@@ -105,4 +114,16 @@ public int getTicksToNextSoulPoint() {
     private void centerSegmentCleared() {
         powderSpecialSegment.replaced();
     }
+
+    @SubscribeEvent
+    public void onChatReceived(ChatMessageReceivedEvent e) {
+        if (!e.getCodedMessage().contains(WYNN_DEATH_MESSAGE)) return;
+        WynntilsMod.postEvent(new CharacterDeathEvent(lastLocation));
+    }
+
+    @SubscribeEvent
+    public void onTick(TickEvent e) {
+        if (McUtils.player() == null) return;
```

Comment:
I guess this works, but I would personally have looked for the player teleport event instead, and only save the last location before being teleported. Then if we get the death message, we know we were teleported since we died.

This is not expensive, but I do want to avoid adding stuff to all ticks, if reasonably possible.

You can keep it this way for now if you just want to get this pushed, or you can implement a mixin for teleportation, if you want to take the opportunity of learning Artemis mixins.

Reply:
Oh, I see that there is already a PlayerTeleportEvent! It is called as "Post" rather than "Pre", though, which means the players current position will be overwritten when you get it. So the mixin should be changed from "RETURN" to "HEAD", and then you can listen to that instead of the TickEvent.